### PR TITLE
Fix shifts

### PR DESCRIPTION
**Pull request by herjy**
This PR fixes a few bugs that were previously affecting the alignment of multiresolution frames. 
* In particular it preforms a needed axis inversion by inverting the sign of the argument of a shift, which is the correct mathematical way of doing it here, instead of inverting the axis of an array.
* The other source for unwelcome shifts was the inconsistency between even and odd shape paddings of images at different resolutions. 
* Another correction that seems to have no effect was to avoid setting arguments of a complex exponential as an axponent. This should generally be avoided as complex exponentials do not behave in the same way as the exponential function in R.

### CODE REVIEW

**Comment by pmelchior** [comment]
Can you check the tests, please

**Comment by herjy** [comment]
shifts were handled inconsistently between the `adam` and the `fix_shifts` branch. Fixed now.

**Comment by pmelchior** [comment]
Please remove the file `docs/hsc_cosmos_35.sca`. It should not be tracked.

**Comment by pmelchior** [comment]
I'll do a few more checks later today, so expect a few comments then.

File: scarlet/resampling.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import numpy as np
 
 
-def match_patches(
-    shape_hr, shape_lr, wcs_hr, wcs_lr, isrot=True, perimeter="overlap", psf=False
-):
+def match_patches(shape_hr, shape_lr, wcs_hr, wcs_lr, isrot = True, perimeter  = 'overlap'):
```
Comment (pmelchior) [inline scarlet/resampling.py]: Can you please rename `perimeter` to `coverage` and `overlap` to intersection.

Comment (pmelchior) [reply scarlet/resampling.py]: This affects several places in this method

File: scarlet/observation.py [hunk 2]
```diff
@@ -434,6 +414,7 @@ def match(self, model_frame):
             model_frame.wcs,
             self.frame.wcs,
             isrot=self.isrot,
+            perimeter = perimeter
```
Comment (pmelchior) [inline scarlet/observation.py]: I don't think this is the way to do it. `perimeter` (or better: `overlap`) needs to affect the construction of the model frame, not the matching of the model frame and the observed frame. The latter really only needs the  intersection plus the width of the diff kernel. This will have to be dealt with in another PR, which also ties in #120. See my code recommendation there.

File: scarlet/observation.py [hunk 3]
```diff
@@ -338,49 +323,34 @@ def sinc_shift(self, imgs, shifts, axes, sum_axis):
...
-            inv_shape,
-            np.array(axes) + len(imgs_shiftfft.shape) - 2,
-        ).image
-
+        # The n-dimensional transform could pose problem for very large images
+        op = fft.Fourier.from_fft(imgs_shiftfft, fft_shape, inv_shape, fft_axes).image
         return op
 
-    def match(self, model_frame):
+    def match(self, model_frame, perimeter = 'union'):
```
Comment (pmelchior) [inline scarlet/observation.py]: See next comment about `perimeter` in `match`

**Comment by pmelchior** [comment]
`LowResObservation.match` seems to have slowed down considerably. I remember taking only 200ms, and it's now around 1s. Do you know the reason? Or am I misremembering?
